### ISSUE TITLE
Problem using bigchaindb-driver with Meteor.js

### ISSUE THREAD

**Issue by perak** [bigchaindb/js-bigchaindb-driver#92]
Simply adding bigchaindb-driver package to Meteor `meteor npm install bigchaindb-driver --save` doesn't work because one of dependencies is using ES6 spread operator (and meteor doesn't transpile npm modules automatically). Namely it's `interledger/five-bells-condition` npm module - see our solution here: 

https://github.com/Integration-Alpha/five-bells-condition/commit/f1d69a637c02869ce45e680bc9020dd9a3225047

So, in order to use bigchaindb with Meteor we forked `js-bigchaindb-driver` and modified it to use our local version of `five-bells-condition`, see here:

https://github.com/Integration-Alpha/js-bigchaindb-driver/commit/2550f6ca6855b8df769c9c27f2a907c7b7276e20

We didn't make a pull request simply because problem is not caused directly by this package (and using local "hacked" version of five-bells-condition is not elegant nor long-term solution), but reporting this in case someone has similar problem or need a quick solution.

So, to use bigchaindb-driver with Meteor, you can use our local "hacked" package:

```
meteor npm install --save https://github.com/Integration-Alpha/js-bigchaindb-driver.git
```

**Comment by TimDaub** [comment]
Interesting. The entry point for "node" is currently set to `"main": "./dist/node/index.js"`. That points to ES6 scripts. Likely we could use one of the bundles we're generating when pushing to npm e.g. https://unpkg.com/bigchaindb-driver@0.3.0/dist/browser/bigchaindb-driver.cjs2.min.js


@vrde @jernejpregelj Makes sense?

**Comment by TimDaub** [comment]
@perak I'm assuming you have Integration-Alpha/js-bigchaindb-driver as a dependency in package.json right now?

Can you try the following:

1. Delete the Integration-Alpha/js-bigchaindb-driver folder in node_modules
2. `npm install bigchaindb-driver`
3. `vim node_modules/bigchaindb-driver/package.json` and change "main" to e.g. `./dist/browser/bigchaindb-driver.cjs2.min.js`
4. Start your meteor build, see if it works and report back here.

If it does, we should think about changing "main" in package.json.

**Comment by perak** [comment]
OK @TimDaub I'll try it now and let you know

**Comment by perak** [comment]
@TimDaub I did it.

Now another error appears, looks like related to pseudo random number generator. `Error: No PRNG` See screenshot:

<img width="1552" alt="screen shot 2017-09-08 at 11 10 47 am" src="https://user-images.githubusercontent.com/5630814/30204673-913f6b38-9486-11e7-875f-7b7f98d216f8.png">

**Comment by perak** [comment]
@TimDaub I am 100 % ready to help, but I believe it will be easier for you if you install meteor and create hello world application, add bigchaindb-driver, paste example code from js-bigchaindb-driver README.md and debug.

To save your time, this few terminal commands will take you there:

```
# Install Meteor
curl https://install.meteor.com/ | sh

# Create "hello world" app
meteor create hello-world

# Install bigchaindb-driver
cd hello-world
meteor npm install bigchaindb-driver --save

# Modify bigchain's package.json as you gave me instructions

# Copy example from this repository's README.md into `hello-world/server/main.js`

# Start the application
meteor

# And you'll see the error.
```

But as I said - I'm here to help, so we can iterate together.

**Comment by TimDaub** [comment]
@perak thanks. I'll have a look. My intuition tell me that the error is raised because five-bells-conditions relies on different crypto-modules for browser and node (as to the best of my knowledge, the crypto APIs of the two are different).

I'll take a look and let you know.

**Comment by TimDaub** [comment]
According to https://github.com/meteor/meteor/issues/8597 we can do two things:

- Remove all ES6 features not compatible with node4 from js-bigchaindb-driver and five-bells-condition
- [Wait until node6 gets LTS as only then meteor upgrades to it](We’ll update to Node 6 once we feel confident that the npm package ecosystem is fully compatible with it, most likely around October 2016 when Node 6 achieves long-term support (LTS) status.)

The root cause being five-bells-condition when used on the meteor server side:
```bash
W20170915-11:49:22.842(2)? (STDERR) /home/tim/ascribe/hello-world/node_modules/five-bells-condition/src/types/prefix-sha256.js:127
W20170915-11:49:22.873(2)? (STDERR)     const subtypes = new Set([...this.subcondition.getSubtypes(), this.subcondition.getTypeName()])
W20170915-11:49:22.874(2)? (STDERR)                               ^^^
W20170915-11:49:22.875(2)? (STDERR) 
W20170915-11:49:22.876(2)? (STDERR) SyntaxError: Unexpected token ...
W20170915-11:49:22.876(2)? (STDERR)     at exports.runInThisContext (vm.js:53:16)
W20170915-11:49:22.877(2)? (STDERR)     at Module._compile (module.js:373:25)
W20170915-11:49:22.877(2)? (STDERR)     at Object.Module._extensions..js (module.js:416:10)
W20170915-11:49:22.878(2)? (STDERR)     at Module.load (module.js:343:32)
W20170915-11:49:22.878(2)? (STDERR)     at Function.Module._load (module.js:300:12)
W20170915-11:49:22.878(2)? (STDERR)     at Module.require (module.js:353:17)
W20170915-11:49:22.879(2)? (STDERR)     at require (internal/module.js:12:17)
W20170915-11:49:22.880(2)? (STDERR)     at Object.<anonymous> (/home/tim/ascribe/hello-world/node_modules/five-bells-condition/index.js:7:22)
W20170915-11:49:22.880(2)? (STDERR)     at Module._compile (module.js:409:26)
W20170915-11:49:22.880(2)? (STDERR)     at Object.Module._extensions..js (module.js:416:10)
=> Exited with code: 1
W20170915-11:49:24.020(2)? (STDERR) /home/tim/ascribe/hello-world/node_modules/five-bells-condition/src/types/prefix-sha256.js:127
W20170915-11:49:24.022(2)? (STDERR)     const subtypes = new Set([...this.subcondition.getSubtypes(), this.subcondition.getTypeName()])
W20170915-11:49:24.023(2)? (STDERR)                               ^^^
W20170915-11:49:24.023(2)? (STDERR) 
W20170915-11:49:24.024(2)? (STDERR) SyntaxError: Unexpected token ...
W20170915-11:49:24.025(2)? (STDERR)     at exports.runInThisContext (vm.js:53:16)
W20170915-11:49:24.026(2)? (STDERR)     at Module._compile (module.js:373:25)
W20170915-11:49:24.026(2)? (STDERR)     at Object.Module._extensions..js (module.js:416:10)
W20170915-11:49:24.026(2)? (STDERR)     at Module.load (module.js:343:32)
W20170915-11:49:24.027(2)? (STDERR)     at Function.Module._load (module.js:300:12)
W20170915-11:49:24.028(2)? (STDERR)     at Module.require (module.js:353:17)
W20170915-11:49:24.029(2)? (STDERR)     at require (internal/module.js:12:17)
W20170915-11:49:24.029(2)? (STDERR)     at Object.<anonymous> (/home/tim/ascribe/hello-world/node_modules/five-bells-condition/index.js:7:22)
W20170915-11:49:24.029(2)? (STDERR)     at Module._compile (module.js:409:26)
W20170915-11:49:24.030(2)? (STDERR)     at Object.Module._extensions..js (module.js:416:10)
=> Exited with code: 1
W20170915-11:49:25.120(2)? (STDERR) /home/tim/ascribe/hello-world/node_modules/five-bells-condition/src/types/prefix-sha256.js:127
W20170915-11:49:25.122(2)? (STDERR)     const subtypes = new Set([...this.subcondition.getSubtypes(), this.subcondition.getTypeName()])
W20170915-11:49:25.122(2)? (STDERR)                               ^^^
W20170915-11:49:25.123(2)? (STDERR) 
W20170915-11:49:25.123(2)? (STDERR) SyntaxError: Unexpected token ...
W20170915-11:49:25.124(2)? (STDERR)     at exports.runInThisContext (vm.js:53:16)
W20170915-11:49:25.125(2)? (STDERR)     at Module._compile (module.js:373:25)
W20170915-11:49:25.125(2)? (STDERR)     at Object.Module._extensions..js (module.js:416:10)
W20170915-11:49:25.126(2)? (STDERR)     at Module.load (module.js:343:32)
W20170915-11:49:25.126(2)? (STDERR)     at Function.Module._load (module.js:300:12)
W20170915-11:49:25.126(2)? (STDERR)     at Module.require (module.js:353:17)
W20170915-11:49:25.127(2)? (STDERR)     at require (internal/module.js:12:17)
W20170915-11:49:25.127(2)? (STDERR)     at Object.<anonymous> (/home/tim/ascribe/hello-world/node_modules/five-bells-condition/index.js:7:22)
W20170915-11:49:25.128(2)? (STDERR)     at Module._compile (module.js:409:26)
W20170915-11:49:25.128(2)? (STDERR)     at Object.Module._extensions..js (module.js:416:10)
=> Exited with code: 1
=> Your application is crashing. Waiting for file change.
```

@perak Is this correct?

**Comment by perak** [comment]
@TimDaub yes, that's correct and PR is already there https://github.com/interledgerjs/five-bells-condition/pull/72

**Comment by jernejpregelj** [comment]
@perak Fixed from 3.1.1
